services/idp/src/index.tsx

```diff
@@ -9,11 +9,18 @@ import store from './store';
 
 import './app.css';
 
+const root = document.getElementById('root')
+
+// if a custome background image has been configured, use it
+const bgImg = root.getAttribute('data-bg-img')
+
 ReactDOM.render(
   <React.StrictMode>
     <Provider store={store as any}>
-      <App/>
+      <div className='oc-login-bg' style={bgImg ? { backgroundImage: `url(${bgImg})` } : {}}>
```

Comment:
Is it possible to have the attributes on the App or even deeper (slot or something), I'd like to avoid adding an extra wrapping div

Reply:
I was hoping to go that route, but I had some trouble getting the background to render when I added this directly on the `App` or children. I'm don't know enough about Material UI to know why that isn't working :/

I made a few adjustments though, I moved the div down to `App` and pass the image url as a prop. I think it's a bit cleaner, keeping the styling contained in `App` and out of the root component.